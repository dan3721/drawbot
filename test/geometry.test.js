// disable logging
process.env.TESTING = true

const geometry = require('../src/geometry')

describe('utility', () => {
  test('distance', () => expect(geometry.distance(3, 3, 6, 7)).toBe(5))
})

describe('shapes', () => {
  test('regularPolygon',
    () => expect(geometry.regularPolygon()).toMatchSnapshot())
  test('circle', () => expect(geometry.circle()).toMatchSnapshot())
  test('square', () => expect(geometry.square()).toMatchSnapshot())
  test('triangle', () => expect(geometry.triangle()).toMatchSnapshot())
  test('star', () => expect(geometry.star()).toMatchSnapshot())
  test('heart', () => expect(geometry.heart()).toMatchSnapshot())
})
