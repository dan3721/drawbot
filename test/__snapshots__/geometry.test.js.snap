// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shapes circle 1`] = `
Object {
  "isCollision": [Function],
  "points": Array [
    Object {
      "x": 2,
      "y": 0,
    },
    Object {
      "x": 1.9900615507308028,
      "y": 0.19913569319163332,
    },
    Object {
      "x": 1.9603449756970877,
      "y": 0.39629228639879516,
    },
    Object {
      "x": 1.9111456115722816,
      "y": 0.5895103488218084,
    },
    Object {
      "x": 1.8429524237408152,
      "y": 0.7768695925493895,
    },
    Object {
      "x": 1.756443146740457,
      "y": 0.9565079572426365,
    },
    Object {
      "x": 1.6524775486319898,
      "y": 1.1266401161272441,
    },
    Object {
      "x": 1.532088886237956,
      "y": 1.2855752193730787,
    },
    Object {
      "x": 1.3964736361721457,
      "y": 1.4317336985194369,
    },
    Object {
      "x": 1.2469796037174672,
      "y": 1.5636629649360596,
    },
    Object {
      "x": 1.0850925277315189,
      "y": 1.680051846301543,
    },
    Object {
      "x": 0.9124213147063258,
      "y": 1.7797436176229373,
    },
    Object {
      "x": 0.73068204873279,
      "y": 1.8617474972884085,
    },
    Object {
      "x": 0.5416809362860103,
      "y": 1.925248493900024,
    },
    Object {
      "x": 0.34729635533386044,
      "y": 1.9696155060244163,
    },
    Object {
      "x": 0.14946018717284834,
      "y": 1.9944075943623603,
    },
    Object {
      "x": -0.049861383476145826,
      "y": 1.9993783640016325,
    },
    Object {
      "x": -0.24868740929497032,
      "y": 1.9844784132003441,
    },
    Object {
      "x": -0.4450418679126287,
      "y": 1.9498558243636472,
    },
    Object {
      "x": -0.6369733005033691,
      "y": 1.8958546923342634,
    },
    Object {
      "x": -0.8225742062612231,
      "y": 1.8230117046233463,
    },
    Object {
      "x": -1.0000000000000004,
      "y": 1.7320508075688772,
    },
    Object {
      "x": -1.16748734446958,
      "y": 1.6238760114317128,
    },
    Object {
      "x": -1.323371675193719,
      "y": 1.4995624059354682,
    },
    Object {
      "x": -1.4661037436596527,
      "y": 1.3603454755418387,
    },
    Object {
      "x": -1.594265014445845,
      "y": 1.2076088206509548,
    },
    Object {
      "x": -1.7065817632643112,
      "y": 1.0428704067589962,
    },
    Object {
      "x": -1.801937735804838,
      "y": 0.8677674782351165,
    },
    Object {
      "x": -1.8793852415718169,
      "y": 0.6840402866513369,
    },
    Object {
      "x": -1.938154572458156,
      "y": 0.49351479538058685,
    },
    Object {
      "x": -1.977661652450257,
      "y": 0.29808453235234855,
    },
    Object {
      "x": -1.9975138424378447,
      "y": 0.09969177132139408,
    },
    Object {
      "x": -1.9975138424378447,
      "y": -0.09969177132139447,
    },
    Object {
      "x": -1.977661652450257,
      "y": -0.29808453235234894,
    },
    Object {
      "x": -1.938154572458156,
      "y": -0.49351479538058723,
    },
    Object {
      "x": -1.8793852415718169,
      "y": -0.6840402866513373,
    },
    Object {
      "x": -1.8019377358048383,
      "y": -0.867767478235116,
    },
    Object {
      "x": -1.706581763264311,
      "y": -1.0428704067589967,
    },
    Object {
      "x": -1.5942650144458448,
      "y": -1.207608820650955,
    },
    Object {
      "x": -1.4661037436596525,
      "y": -1.360345475541839,
    },
    Object {
      "x": -1.3233716751937188,
      "y": -1.4995624059354684,
    },
    Object {
      "x": -1.1674873444695797,
      "y": -1.623876011431713,
    },
    Object {
      "x": -0.9999999999999992,
      "y": -1.7320508075688776,
    },
    Object {
      "x": -0.8225742062612232,
      "y": -1.8230117046233463,
    },
    Object {
      "x": -0.6369733005033682,
      "y": -1.8958546923342636,
    },
    Object {
      "x": -0.4450418679126292,
      "y": -1.9498558243636472,
    },
    Object {
      "x": -0.2486874092949699,
      "y": -1.9844784132003441,
    },
    Object {
      "x": -0.04986138347614631,
      "y": -1.9993783640016325,
    },
    Object {
      "x": 0.14946018717284873,
      "y": -1.9944075943623603,
    },
    Object {
      "x": 0.34729635533386166,
      "y": -1.969615506024416,
    },
    Object {
      "x": 0.5416809362860102,
      "y": -1.925248493900024,
    },
    Object {
      "x": 0.7306820487327907,
      "y": -1.8617474972884083,
    },
    Object {
      "x": 0.9124213147063257,
      "y": -1.7797436176229373,
    },
    Object {
      "x": 1.0850925277315193,
      "y": -1.6800518463015426,
    },
    Object {
      "x": 1.2469796037174667,
      "y": -1.5636629649360598,
    },
    Object {
      "x": 1.396473636172146,
      "y": -1.4317336985194367,
    },
    Object {
      "x": 1.5320888862379567,
      "y": -1.2855752193730778,
    },
    Object {
      "x": 1.6524775486319898,
      "y": -1.126640116127244,
    },
    Object {
      "x": 1.7564431467404575,
      "y": -0.9565079572426356,
    },
    Object {
      "x": 1.8429524237408152,
      "y": -0.7768695925493895,
    },
    Object {
      "x": 1.9111456115722816,
      "y": -0.5895103488218078,
    },
    Object {
      "x": 1.9603449756970877,
      "y": -0.39629228639879543,
    },
    Object {
      "x": 1.9900615507308028,
      "y": -0.19913569319163282,
    },
  ],
}
`;

exports[`shapes heart 1`] = `
Object {
  "points": Array [
    Object {
      "x": 0,
      "y": -3.141592653589793,
    },
    Object {
      "x": -0.1,
      "y": -2.6905658417935308,
    },
    Object {
      "x": -0.2,
      "y": -2.498091544796509,
    },
    Object {
      "x": -0.3,
      "y": -2.3461938234056494,
    },
    Object {
      "x": -0.4,
      "y": -2.214297435588181,
    },
    Object {
      "x": -0.5,
      "y": -2.0943951023931953,
    },
    Object {
      "x": -0.6,
      "y": -1.9823131728623846,
    },
    Object {
      "x": -0.7,
      "y": -1.8754889808102941,
    },
    Object {
      "x": -0.8,
      "y": -1.7721542475852272,
    },
    Object {
      "x": -0.9,
      "y": -1.6709637479564563,
    },
    Object {
      "x": -1,
      "y": -1.5707963267948966,
    },
    Object {
      "x": -1.1,
      "y": -1.4706289056333366,
    },
    Object {
      "x": -1.2,
      "y": -1.3694384060045657,
    },
    Object {
      "x": -1.3,
      "y": -1.266103672779499,
    },
    Object {
      "x": -1.4,
      "y": -1.1592794807274085,
    },
    Object {
      "x": -1.5,
      "y": -1.0471975511965974,
    },
    Object {
      "x": -1.6,
      "y": -0.9272952180016119,
    },
    Object {
      "x": -1.7,
      "y": -0.7953988301841433,
    },
    Object {
      "x": -1.8,
      "y": -0.6435011087932843,
    },
    Object {
      "x": -1.9,
      "y": -0.45102681179626236,
    },
    Object {
      "x": -2,
      "y": 0,
    },
    Object {
      "x": -2,
      "y": 0,
    },
    Object {
      "x": -1.9,
      "y": 0.43588989435406755,
    },
    Object {
      "x": -1.8,
      "y": 0.5999999999999999,
    },
    Object {
      "x": -1.7,
      "y": 0.714142842854285,
    },
    Object {
      "x": -1.6,
      "y": 0.7999999999999999,
    },
    Object {
      "x": -1.5,
      "y": 0.8660254037844386,
    },
    Object {
      "x": -1.4,
      "y": 0.9165151389911681,
    },
    Object {
      "x": -1.3,
      "y": 0.9539392014169457,
    },
    Object {
      "x": -1.2,
      "y": 0.9797958971132712,
    },
    Object {
      "x": -1.1,
      "y": 0.99498743710662,
    },
    Object {
      "x": -1,
      "y": 1,
    },
    Object {
      "x": -0.9,
      "y": 0.99498743710662,
    },
    Object {
      "x": -0.8,
      "y": 0.9797958971132712,
    },
    Object {
      "x": -0.7,
      "y": 0.9539392014169457,
    },
    Object {
      "x": -0.6,
      "y": 0.916515138991168,
    },
    Object {
      "x": -0.5,
      "y": 0.8660254037844386,
    },
    Object {
      "x": -0.4,
      "y": 0.8,
    },
    Object {
      "x": -0.3,
      "y": 0.714142842854285,
    },
    Object {
      "x": -0.2,
      "y": 0.5999999999999999,
    },
    Object {
      "x": -0.1,
      "y": 0.4358898943540673,
    },
    Object {
      "x": 0,
      "y": 0,
    },
    Object {
      "x": 0,
      "y": 0,
    },
    Object {
      "x": 0.1,
      "y": 0.4358898943540673,
    },
    Object {
      "x": 0.2,
      "y": 0.5999999999999999,
    },
    Object {
      "x": 0.3,
      "y": 0.714142842854285,
    },
    Object {
      "x": 0.4,
      "y": 0.8,
    },
    Object {
      "x": 0.5,
      "y": 0.8660254037844386,
    },
    Object {
      "x": 0.6,
      "y": 0.916515138991168,
    },
    Object {
      "x": 0.7,
      "y": 0.9539392014169457,
    },
    Object {
      "x": 0.8,
      "y": 0.9797958971132712,
    },
    Object {
      "x": 0.9,
      "y": 0.99498743710662,
    },
    Object {
      "x": 1,
      "y": 1,
    },
    Object {
      "x": 1.1,
      "y": 0.99498743710662,
    },
    Object {
      "x": 1.2,
      "y": 0.9797958971132712,
    },
    Object {
      "x": 1.3,
      "y": 0.9539392014169457,
    },
    Object {
      "x": 1.4,
      "y": 0.9165151389911681,
    },
    Object {
      "x": 1.5,
      "y": 0.8660254037844386,
    },
    Object {
      "x": 1.6,
      "y": 0.7999999999999999,
    },
    Object {
      "x": 1.7,
      "y": 0.714142842854285,
    },
    Object {
      "x": 1.8,
      "y": 0.5999999999999999,
    },
    Object {
      "x": 1.9,
      "y": 0.43588989435406755,
    },
    Object {
      "x": 2,
      "y": 0,
    },
    Object {
      "x": 2,
      "y": 0,
    },
    Object {
      "x": 1.9,
      "y": -0.45102681179626236,
    },
    Object {
      "x": 1.8,
      "y": -0.6435011087932843,
    },
    Object {
      "x": 1.7,
      "y": -0.7953988301841433,
    },
    Object {
      "x": 1.6,
      "y": -0.9272952180016119,
    },
    Object {
      "x": 1.5,
      "y": -1.0471975511965974,
    },
    Object {
      "x": 1.4,
      "y": -1.1592794807274085,
    },
    Object {
      "x": 1.3,
      "y": -1.266103672779499,
    },
    Object {
      "x": 1.2,
      "y": -1.3694384060045657,
    },
    Object {
      "x": 1.1,
      "y": -1.4706289056333366,
    },
    Object {
      "x": 1,
      "y": -1.5707963267948966,
    },
    Object {
      "x": 0.9,
      "y": -1.6709637479564563,
    },
    Object {
      "x": 0.8,
      "y": -1.7721542475852272,
    },
    Object {
      "x": 0.7,
      "y": -1.8754889808102941,
    },
    Object {
      "x": 0.6,
      "y": -1.9823131728623846,
    },
    Object {
      "x": 0.5,
      "y": -2.0943951023931953,
    },
    Object {
      "x": 0.4,
      "y": -2.214297435588181,
    },
    Object {
      "x": 0.3,
      "y": -2.3461938234056494,
    },
    Object {
      "x": 0.2,
      "y": -2.498091544796509,
    },
    Object {
      "x": 0.1,
      "y": -2.6905658417935308,
    },
    Object {
      "x": 0,
      "y": -3.141592653589793,
    },
  ],
  "sCollision": [Function],
}
`;

exports[`shapes regularPolygon 1`] = `
Object {
  "isCollision": [Function],
  "points": Array [
    Object {
      "x": 2,
      "y": 0,
    },
    Object {
      "x": 0.6180339887498949,
      "y": 1.902113032590307,
    },
    Object {
      "x": -1.6180339887498947,
      "y": 1.1755705045849465,
    },
    Object {
      "x": -1.618033988749895,
      "y": -1.175570504584946,
    },
    Object {
      "x": 0.6180339887498945,
      "y": -1.9021130325903073,
    },
  ],
}
`;

exports[`shapes square 1`] = `
Object {
  "isCollision": [Function],
  "points": Array [
    Object {
      "x": -1,
      "y": 1,
    },
    Object {
      "x": 1,
      "y": 1,
    },
    Object {
      "x": 1,
      "y": -1,
    },
    Object {
      "x": -1,
      "y": -1,
    },
  ],
}
`;

exports[`shapes star 1`] = `
Object {
  "isCollision": [Function],
  "points": Array [
    Object {
      "x": 0.37082039324993693,
      "y": -1.141267819554184,
    },
    Object {
      "x": 0.38832815729997483,
      "y": -0.2821369211003871,
    },
    Object {
      "x": 1.2,
      "y": 0,
    },
    Object {
      "x": 0.38832815729997483,
      "y": 0.2821369211003871,
    },
    Object {
      "x": 0.37082039324993693,
      "y": 1.141267819554184,
    },
    Object {
      "x": -0.14832815729997473,
      "y": 0.45650712782167374,
    },
    Object {
      "x": -0.9708203932499367,
      "y": 0.7053423027509679,
    },
    Object {
      "x": -0.48,
      "y": 5.878304635907296e-17,
    },
    Object {
      "x": -0.9708203932499369,
      "y": -0.7053423027509677,
    },
    Object {
      "x": -0.14832815729997484,
      "y": -0.4565071278216737,
    },
  ],
}
`;

exports[`shapes triangle 1`] = `
Object {
  "isCollision": [Function],
  "points": Array [
    Object {
      "x": 0,
      "y": 1,
    },
    Object {
      "x": 1,
      "y": -1,
    },
    Object {
      "x": -1,
      "y": -1,
    },
  ],
}
`;
